app/test/vesting_test.go

```diff
@@ -0,0 +1,936 @@
+		getAddress(name, s.cctx.Keyring),
+		validators[0].GetOperator(),
+		sdk.NewCoin(app.BondDenom, sdk.NewInt(testfactory.BaseAccountDefaultBalance)),
+	)
+	resTx, err := testnode.SignAndBroadcastTx(s.ecfg, s.cctx.Context, name, []sdk.Msg{msgDelg}...)
+	assert.NoError(s.T(), err)
+
+	resQ, err := s.cctx.WaitForTx(resTx.TxHash, 10)
+	assert.NoError(s.T(), err)
+	assert.EqualValues(s.T(), 0, resQ.TxResult.Code, "the delegation TX must succeed")
+
+	// verify the delegations
+	del, err = GetAccountDelegations(s.cctx.GRPCClient, address)
+	assert.NoError(s.T(), err)
+	assert.NotEmpty(s.T(), del, "delegations must not be empty")
+	assert.EqualValues(s.T(),
+		testfactory.BaseAccountDefaultBalance,
+		del[0].Balance.Amount.Int64(),
+		"delegation amount must match")
+}
+
+// testClaimDelegationReward tests the claiming of delegation rewards for a vesting account.
+// It takes the name of the vesting account (name) as an input.
+// It claims the delegation rewards and then retrieves the balances of the vesting account.
+// It asserts that the balance after claiming the reward.
+func (s *VestingModuleTestSuite) testClaimDelegationReward(name string) {
+	assert.NoError(s.T(), s.cctx.WaitForNextBlock())
+
+	address := getAddress(name, s.cctx.Keyring).String()
+
+	cli := distributiontypes.NewQueryClient(s.cctx.GRPCClient)
+	resR, err := cli.DelegationTotalRewards(
+		context.Background(),
+		&distributiontypes.QueryDelegationTotalRewardsRequest{
+			DelegatorAddress: address,
+		},
+	)
+	assert.NoError(s.T(), err)
+	assert.NotEmpty(s.T(), resR, "empty staking rewards")
+	assert.NotEmpty(s.T(), resR.Rewards)
+
+	rewardAmount := resR.Rewards[0].Reward.AmountOf(app.BondDenom).RoundInt().Int64()
+	assert.Greater(s.T(), rewardAmount, int64(0), "rewards must be more than zero")
+
+	balancesBefore, err := GetAccountSpendableBalance(s.cctx.GRPCClient, address)
+	assert.NoError(s.T(), err)
+
+	// minExpectedBalance is used because more tokens may be vested to the
+	// account in the middle of this test
+	minExpectedBalance := balancesBefore.AmountOf(app.BondDenom).Int64() + rewardAmount
+
+	validators, err := GetValidators(s.cctx.GRPCClient)
+	assert.NoError(s.T(), err)
+	assert.NotEmpty(s.T(), validators, "empty validators set")
+
+	msg := distributiontypes.NewMsgWithdrawDelegatorReward(
+		getAddress(name, s.cctx.Keyring),
+		validators[0].GetOperator(),
+	)
+	resTx, err := testnode.SignAndBroadcastTx(s.ecfg, s.cctx.Context, name, []sdk.Msg{msg}...)
+	assert.NoError(s.T(), err)
+
+	resQ, err := s.cctx.WaitForTx(resTx.TxHash, 10)
+	assert.NoError(s.T(), err)
+	assert.EqualValues(s.T(), 0, resQ.TxResult.Code, "the claim reward TX must succeed")
+
+	// Check if the reward amount in the account
+	balancesAfter, err := GetAccountSpendableBalance(s.cctx.GRPCClient, address)
+	assert.NoError(s.T(), err)
+
+	assert.GreaterOrEqual(s.T(), balancesAfter.AmountOf(app.BondDenom).Int64(), minExpectedBalance, "Minimum balance after claiming reward")
```

Comment:
[nit] I don't think the error message provides any additional information

```suggestion
	assert.GreaterOrEqual(s.T(), balancesAfter.AmountOf(app.BondDenom).Int64(), minExpectedBalance)
```